internal/connectors/plugins/public/currencycloud/client/client.go

```diff
@@ -8,64 +8,68 @@ import (
 
 	"github.com/formancehq/payments/internal/connectors/httpwrapper"
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"golang.org/x/sync/singleflight"
 )
 
+//go:generate mockgen -source client.go -destination client_generated.go -package client . Client
+type Client interface {
+	GetAccounts(ctx context.Context, page int, pageSize int) ([]*Account, int, error)
+	GetBalances(ctx context.Context, page int, pageSize int) ([]*Balance, int, error)
+	GetBeneficiaries(ctx context.Context, page int, pageSize int) ([]*Beneficiary, int, error)
+	GetContactID(ctx context.Context, accountID string) (*Contact, error)
+	GetTransactions(ctx context.Context, page int, pageSize int, updatedAtFrom time.Time) ([]Transaction, int, error)
+}
+
 type apiTransport struct {
-	authToken  string
+	c          *client
 	underlying *otelhttp.Transport
 }
 
 func (t *apiTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	req.Header.Add("X-Auth-Token", t.authToken)
+	if t.c.authToken != "" {
+		req.Header.Add("X-Auth-Token", t.c.authToken)
+	}
 
 	return t.underlying.RoundTrip(req)
 }
 
-type Client struct {
+type client struct {
 	httpClient httpwrapper.Client
 	endpoint   string
 	loginID    string
 	apiKey     string
+
+	singleFlight singleflight.Group
+	authToken    string
 }
 
-func (c *Client) buildEndpoint(path string, args ...interface{}) string {
+func (c *client) buildEndpoint(path string, args ...interface{}) string {
 	return fmt.Sprintf("%s/%s", c.endpoint, fmt.Sprintf(path, args...))
 }
 
 const DevAPIEndpoint = "https://devapi.currencycloud.com"
 
-func newHTTPClient() *http.Client {
+func NewHTTPClient() *http.Client {
```

Comment:
I'm surprised the linter isn't saying anything but this appears to be unused now that you removed the authentication call at install time.